### PR TITLE
Add filtering for Project Scorecard reports

### PR DESCRIPTION
**Pull request by gigxz**
Adds filtering to Project Scorecard history page. Users with can-view-all-reports permission can filter by report author. By default, it shows the last 90 days.

The old `for_project` and `for_project_group` routes are replaced with links to the history page, with project filters applied. This link defaults to showing the past year.

---

Screen Shot 2022-04-08 at 11 44 06 AM

Tested
- [x] With confidential project names
- [x] With user that's not permitted to see projects
- [x] With user that's not permitted to see reports run by other users

### CODE REVIEW

File: drivers/project_scorecard/app/controllers/project_scorecard/warehouse_reports/scorecards_controller.rb [hunk 1]
```diff
@@ -215,6 +233,36 @@ def project_group_scope
...
+
+      project_group_ids = @history_filter.project_group_ids
+      scope = scope.where(project_group: project_group_ids.uniq) if project_group_ids&.any?
+
+      scope
+    end
+
+    private def set_history_filter
+      # user_id is used to filter what projects are listed in the dropdown, so it has to be set to curr user
+      # TODO look into using a custom filter alongside filter base..
```
Comment (gigxz) [inline drivers/project_scorecard/app/controllers/project_scorecard/warehouse_reports/scorecards_controller.rb]: need to figure out how to filter by report author

File: drivers/project_scorecard/app/controllers/project_scorecard/warehouse_reports/scorecards_controller.rb [hunk 2]
```diff
@@ -215,6 +233,36 @@ def project_group_scope
       @filter = ::Filters::FilterBase.new(initial_filter_params.merge(user_id: current_user.id, project_type_codes: []))
     end
 
+    private def filtered_reports_scope
+      scope = reports_scope.started_between(start_date: @history_filter.start, end_date: @history_filter.end)
+
+      scope = scope.where(project_id: project_scope.select(:id)).
+        or(scope.where(project_group_id: project_group_scope.select(:id)))
```
Comment (gigxz) [inline drivers/project_scorecard/app/controllers/project_scorecard/warehouse_reports/scorecards_controller.rb:214]: These are filtering for viewability. We only really need to do it if `@history_filter.project_ids` and `@history_filter.project_group_ids` are both empty, I think? (Although even if they are filled, is it possible they could include a report that is not viewable, by messing with query params? )

**Comment by gigxz** [comment]
@eanders for users that are only allowed to view assigned projects, the behavior I'm seeing is: the dropdowns show all projects, but the results table correctly only shows reports for the project that the user is assigned to. Is that the correct behavior, or is something going wrong with the dropdowns?

File: drivers/project_scorecard/app/views/project_scorecard/warehouse_reports/scorecards/_report_list.haml [hunk 3]
```diff
@@ -1,29 +1,28 @@
-- if @reports.exists?
```
Comment (eanders) [inline drivers/project_scorecard/app/views/project_scorecard/warehouse_reports/scorecards/_report_list.haml:1]: It looks like we lost the "No reports on file" text when there are no reports to show.  Will it just show an empty table?  If that's the case, can we bring that back?  Maybe it just moved up the chain, but I'm not seeing it.

File: drivers/project_scorecard/app/views/project_scorecard/warehouse_reports/scorecards/_report_list.haml [hunk 4]
```diff
@@ -1,29 +1,28 @@
...
-          %th Date Range
-          %th Run On
-          %th Completed On
-          %th Author
-      %tbody
-        - @reports.each do |report|
+.table-utilities
+%p= page_entries_info @reports, entry_name: _('scorecard')
+.row
+  .col-sm-12
```
Comment (eanders) [inline drivers/project_scorecard/app/views/project_scorecard/warehouse_reports/scorecards/_report_list.haml:4]: This should be unnecessary, was it behaving oddly when not wrapped?

**Review by awisema — APPROVED** [review]

File: drivers/project_scorecard/app/views/project_scorecard/warehouse_reports/scorecards/_report_list.haml (hunk 4 above)
Comment (gigxz) [reply drivers/project_scorecard/app/views/project_scorecard/warehouse_reports/scorecards/_report_list.haml:4]: Hmm nope it's not necessary, just a hasty copy-paste. I can remove.

File: drivers/project_scorecard/app/views/project_scorecard/warehouse_reports/scorecards/_report_list.haml (hunk 3 above)
Comment (gigxz) [reply drivers/project_scorecard/app/views/project_scorecard/warehouse_reports/scorecards/_report_list.haml:1]: The `page_entries_info` line does it.

On that note, we had the conversation about moving to Pagy for the other DQ report ticket, but I didn't update this one. I can do it in a follow-up PR.

Screen Shot 2022-04-11 at 8 37 32 AM

File: drivers/project_scorecard/app/views/project_scorecard/warehouse_reports/scorecards/_report_list.haml (hunk 4 above)
Comment (eanders) [reply drivers/project_scorecard/app/views/project_scorecard/warehouse_reports/scorecards/_report_list.haml:4]: Thanks!

File: drivers/project_scorecard/app/views/project_scorecard/warehouse_reports/scorecards/_report_list.haml (hunk 3 above)
Comment (eanders) [reply drivers/project_scorecard/app/views/project_scorecard/warehouse_reports/scorecards/_report_list.haml:1]: That's probably fine.  In other places, we go ahead and hide the table header, it just looks cleaner, but I'm not sure it's important enough to implement here.

Comment (gigxz) [reply drivers/project_scorecard/app/views/project_scorecard/warehouse_reports/scorecards/_report_list.haml:1]: That's definitely cleaner, and easy enough to do : https://github.com/greenriver/hmis-warehouse/pull/2134

File: drivers/project_scorecard/app/views/project_scorecard/warehouse_reports/scorecards/_report_list.haml (hunk 4 above)
Comment (gigxz) [reply drivers/project_scorecard/app/views/project_scorecard/warehouse_reports/scorecards/_report_list.haml:4]: https://github.com/greenriver/hmis-warehouse/pull/2134

**Pull request merged**